server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -981,28 +981,19 @@ public ReplicaResponse(long localCheckpoint, long globalCheckpoint) {
         @Override
         public void readFrom(StreamInput in) throws IOException {
             super.readFrom(in);
-            if (in.getVersion().onOrAfter(Version.V_6_0_0_alpha1)) {
-                localCheckpoint = in.readZLong();
-            } else {
-                // 5.x used to read empty responses, which don't really read anything off the stream, so just do nothing.
-                localCheckpoint = SequenceNumbers.PRE_60_NODE_CHECKPOINT;
-            }
-            if (in.getVersion().onOrAfter(Version.V_6_0_0_rc1)) {
-                globalCheckpoint = in.readZLong();
-            } else {
-                globalCheckpoint = SequenceNumbers.PRE_60_NODE_CHECKPOINT;
-            }
+            assert in.getVersion().onOrAfter(Version.V_6_0_0_alpha1);
```

Comment:
The later (stronger) assertion should cover this assertion. I am not sure about the usefulness of these assertions.

Reply:
Yeah I know what you mean. I want to have _something_ here for documentation and to save the heartache that arises from tracking down serialisation issues, even though in theory we should never hit these. And then I felt the need to double-check that `rc1` was definitely a tighter check. I'll drop the looser one.